examples/with-iron-session/pages/profile-ssr.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react'
+import Layout from '../components/layout'
+import withSession from '../lib/session'
+import PropTypes from 'prop-types'
+
+const SsrProfile = ({ user }) => {
+  return (
+    <Layout>
+      <h1>Your GitHub profile</h1>
+      <h2>
+        This page uses{' '}
+        <a href="https://nextjs.org/docs/basic-features/pages#server-side-rendering">
+          Server-side Rendering (SSR)
+        </a>{' '}
+        and{' '}
+        <a href="https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering">
+          getServerSideProps
+        </a>
+      </h2>
+
+      {user?.isLoggedIn && (
+        <>
+          <p style={{ fontStyle: 'italic' }}>
+            Public data, from{' '}
+            <a href={githubUrl(user.login)}>{githubUrl(user.login)}</a>, reduced
+            to `login` and `avatar_url`.
+          </p>
+          <pre>{JSON.stringify(user, undefined, 2)}</pre>
+        </>
+      )}
+    </Layout>
+  )
+}
+
+export const getServerSideProps = withSession(async function({ req, res }) {
+  const user = req.session.get('user')
+
+  if (user === undefined) {
+    res.setHeader('location', '/login')
+    res.statusCode = 302
+    res.end()
+    return
+  }
```

Comment:
I'm not convinced that this is a good pattern. See [my observations](https://github.com/zeit/next.js/discussions/11281?sort_order=chronological#discussioncomment-12067) about performing redirects in `getServerSideProps`. The fact that it works is happenstance and it's quite inefficient.

Reply:
@justincy Thanks for the comment, if you happen to find a better way to redirect from getServerSideProps then I'll welcome the contribution at https://github.com/vvo/next-iron-session/, thanks :)